### PR TITLE
Extract track cover image from the file, if it has one

### PR DESCRIPTION
**Pull request by vnord**
This is my first time writing Android code, and my first "serious" PR, so be gentle if I did anything stupid :)

### CODE REVIEW

File: app/src/main/java/com/prangesoftwaresolutions/audioanchor/PlayActivity.java [hunk 1]
```diff
@@ -385,7 +388,25 @@ void setAlbumCover() {
                 display.getSize(size);
                 reqSize = java.lang.Math.max(size.x, size.y);
             }
-            BitmapUtils.setImage(mCoverIV, mAudioFile.getCoverPath(), reqSize);
+
+            // Taken from https://stackoverflow.com/a/21549403
+            android.media.MediaMetadataRetriever mmr = new MediaMetadataRetriever();
```
Comment (flackbash) [inline app/src/main/java/com/prangesoftwaresolutions/audioanchor/PlayActivity.java]: just use "MediaMetadataRetriever mmr = new MediaMetadataRetriever();"

**Comment by flackbash** [comment]
Great! :) I thought about adding this but got stuck on the thought which cover to show in the MainActivity (the list of albums) and in the AlbumActivity (the list of audio files), since audio files within the same album directory can have different cover images in their meta data. Also, it might be weird to have the cover of the audio file be different from the cover that is shown for the album. But I think it's good if we do it as you implemented it and then add an option in the settings to use the image file in the directory as a cover in the PlayActivity.

There are just two small things that I would add/change about this PR:
1) see the comment in the commit
2) the notification should show the same cover image as is shown in the PlayActivity. To change this, it should suffice to change which image is assigned to notificationCover [here](https://github.com/flackbash/AudioAnchor/blob/39120a72ae197c2f3bdb67184c44212cc3315c3a/app/src/main/java/com/prangesoftwaresolutions/audioanchor/MediaPlayerService.java#L539)

If you are not sure how to apply the change for 1) and keep the git history clean maybe @jubalh can help. I'm no git expert myself.

I can add 2) myself but if you like, feel free to implement it :)

File: app/src/main/java/com/prangesoftwaresolutions/audioanchor/PlayActivity.java (hunk 1 above)
Comment (vnord) [reply app/src/main/java/com/prangesoftwaresolutions/audioanchor/PlayActivity.java]: Obviously. Sorry! I think the latest commit of the branch should have taken care of that.

**Comment by jubalh** [comment]
Since your second commit was just fixing your first commit, you could have used `git commit --amend; git push --force`.
Now that you have two commits for a actually one change you should squash your commit. `git rebase -i HEAD~2` and then use `s` for the second commit.

**Comment by vnord** [comment]
I tried fixing the git history, but I think I may have just made it worse, sorry! I'll try to clear things when I get back from dinner.

I already implemented 2), so if you think the approach in this PR is all right, I'd be happy to commit that in a bit. I'd also be happy to look into adding a setting for using the image in the directory as the cover in the PlayActivity.

**Comment by jubalh** [comment]
> I tried fixing the git history, but I think I may have just made it worse, sorry! I'll try to clear things when I get back from dinner.

Yes it's worse :-)
Maybe this helps: https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history

**Comment by jubalh** [comment]
Well done. Keep in mind that you seem to have chose an email not registered with your GitHub account. That's why the icon with the commit is grey instead of your user icon.

**Comment by vnord** [comment]
> Keep in mind that you seem to have chose an email not registered with your GitHub account. That's why the icon with the commit is grey instead of your user icon.

Weird, I've been working on the same computer the entire time, and I double checked that I was using the same address as the one indicated on Github. Is it a problem, and should I attempt to do something about it retroactively?

**Comment by jubalh** [comment]
It's not a problem. But it means the contribution will not show on your GitHub profile. So it depends on you if you want to change it or not.
You can configure the email/name in you `.gitconfig`. Globally or locally. Also see the `git config` command.

**Comment by vnord** [comment]
>  You can configure the email/name in you `.gitconfig`. Globally or locally. Also see the `git config` command.

Everything looked correct in `.gitconfig` before the last commit, yet it still didn't register properly. But oh well, I guess it's not a big deal.

**Comment by vnord** [comment]
Umm, guess my attempt at fixing my username did not go so well :( It seems the histories are now unrelated and that github automatically closed the PR somehow. Any idea what I should do in order to not mess it up further? Sorry :(!

**Comment by vnord** [comment]
Guess I was extremely stupid and just copy-pasted someone's snippet which rewrote all emails+usernames and not just mine in the history. Should I copy the changes and just delete this PR and branch, and make a new one and open a PR for that?

**Comment by jubalh** [comment]
You dont need to create a new PR. You can remove your local directory, then clone again from upstream, then create again a branch named track-cover-from-file, then apply your changes, and then force push to GitHub ;)

Maybe reopen the PR first.

**Comment by vnord** [comment]
> You dont need to create a new PR. You can remove your local directory, then clone again from upstream, then create again a branch named track-cover-from-file, then apply your changes, and then force push to GitHub ;)
> 
> Maybe reopen the PR first.

Hm, the history looks all right now, but this PR doesn't seem to be aware of that, and it won't let me reopen it due to "the track-cover-from-file-branch was force-pushed or recreated".

**Comment by vnord** [comment]
I'll just open a new PR, as I have to head off to bed now. Thanks a lot for all the help, though, and sorry for committing so many git crimes - I'll try my best to better my ways.

**Pull request closed without merging**